src/platform/android/AndroidChipPlatform-JNI.cpp

```diff
@@ -124,6 +126,46 @@ JNI_METHOD(void, initChipStack)(JNIEnv * env, jobject self)
     VerifyOrReturn(err == CHIP_NO_ERROR, ChipLogError(DeviceLayer, "Error initializing CHIP stack: %s", ErrorStr(err)));
 }
 
+// for NFCCommissioningManager
+JNI_METHOD(void, nativeSetNFCCommissioningManager)(JNIEnv * env, jobject, jobject manager)
+{
+    ChipLogProgress(DeviceLayer, "(Android JNI) nativeSetNFCCommissioningManager()");
+
+    VerifyOrReturn(env != nullptr);
+#if CHIP_DEVICE_CONFIG_ENABLE_NFC_BASED_COMMISSIONING
+    chip::DeviceLayer::StackLock lock;
+    chip::DeviceLayer::Internal::NFCCommissioningMgrImpl().InitializeWithObject(manager);
+#endif // CHIP_DEVICE_CONFIG_ENABLE_NFC_BASED_COMMISSIONING
+}
+
+JNI_METHOD(void, onNfcTagResponse)(JNIEnv * env, jobject self, jbyteArray jbArray)
+{
+    ChipLogProgress(Controller, "%p (Android JNI) onNfcTagResponse()", self);
+
+    VerifyOrReturn(env != nullptr);
+#if CHIP_DEVICE_CONFIG_ENABLE_NFC_BASED_COMMISSIONING
```

Comment:
please add JniLocalReferenceScope scope(env); for the jni callback function, for example, 
https://github.com/project-chip/connectedhomeip/blob/ff458d012d392bdf4d5cef0bbee56b5b0fe83c74/src/controller/java/AndroidDeviceControllerWrapper.cpp#L808